xmodule/library_tools.py

```diff
@@ -27,39 +43,63 @@ def normalize_key_for_search(library_key):
 
 class LibraryToolsService:
     """
-    Service that allows LibraryContentBlock to interact with libraries in the
-    modulestore.
+    Service for LibraryContentBlock and LibrarySourcedBlock.
+
+    Allows to interact with libraries in the modulestore and blockstore.
     """
     def __init__(self, modulestore, user_id):
         self.store = modulestore
         self.user_id = user_id
 
-    def _get_library(self, library_key):
+    def _get_library(self, library_key, is_v2_lib):
```

Comment:
Nit: it's often ambiguous whether parameters like `library_key` here are accepting a library key or a string. I would suggest adding the explicit type hint `library_key: LearningContextKey` declaration.

And it seems odd to accept `is_v2_lib` as a parameter when it can be trivially deduced from the key - 

Instead of `if is_v2_lib:` use `if isinstance(lib_key, LibraryLocatorV2):`

Reply:
https://github.com/openedx/edx-platform/pull/33263#pullrequestreview-1627813331